opendevin/core/main.py

```diff
@@ -110,9 +115,8 @@ async def on_event(event: Event):
         await asyncio.sleep(1)  # Give back control for a tick, so the agent can run
 
     # retrieve the final state before we close the controller and agent
-    final_agent_state = controller.get_agent_state()
-    await controller.close()
-    return final_agent_state
+    final_state = await controller.close()
+    return final_state
```

Comment:
`controller.close()` doesn't seem to return a state, and I don't think we want it to. Maybe you just mean `return controller.state`?

Reply:
Fixed! How about now!